flink-table/flink-sql-client/src/main/java/org/apache/flink/table/client/gateway/local/ExecutionContext.java

```diff
@@ -428,6 +485,12 @@ private void initializeTableEnvironment() {
 			modules.forEach(tableEnv::loadModule);
 		}
 
+		if (catalogManager != null) {
+			registerFunctions();
```

Comment:
why register function if we already have the catalog manager?

Reply:
No need to register because we also book keep the FunctionCatalog now.